src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -219,30 +225,44 @@ struct nonmaxsuppression
 
     // filter boxes below score_threshold
     template <class T>
-    std::priority_queue<std::pair<double, int64_t>>
+    std::vector<std::pair<double, int64_t>>
     filter_boxes_by_score(T scores_start, std::size_t num_boxes, double score_threshold) const
     {
-        std::priority_queue<std::pair<double, int64_t>> boxes_heap;
-        auto insert_to_boxes_heap =
-            make_function_output_iterator([&](const auto& x) { boxes_heap.push(x); });
+        std::vector<std::pair<double, int64_t>> boxes_heap;
         int64_t box_idx = 0;
-        transform_if(
-            scores_start,
-            scores_start + num_boxes,
-            insert_to_boxes_heap,
-            [&](auto sc) {
-                box_idx++;
-                return sc >= score_threshold;
-            },
-            [&](auto sc) { return std::make_pair(sc, box_idx - 1); });
+
+        if(score_threshold > 0.0)
+        {
+            transform_if(
+                scores_start,
+                scores_start + num_boxes,
+                std::back_inserter(boxes_heap),
+                [&](auto sc) {
+                    box_idx++;
+                    return sc >= score_threshold;
+                },
+                [&](auto sc) { return std::make_pair(sc, box_idx - 1); });
+        }
+        else
+        { // score is irrelevant, just push into boxes_heap and make a score-index pair
+            std::transform(scores_start,
```

Comment:
why do we not use `execution` policy here as well ?

Reply:
The use of box_idx++ here thats then used for box_idx - 1 having shared state between runs as the idx, is tied to a localtion in the box list of 4 points it needs to grab later when we batch_box

Reply:
Also we are using `std::back_inserter` which probably wont be parallelized.